### PR TITLE
Refactor to support Android

### PR DESCRIPTION
**Pull request by skevy**
This PR adds Android support (powered by ExponentFontLoader in Exponent), and also refactors so WithCustomFont is a universal component, and the font loader code itself is platform specific.

### CODE REVIEW

File: utils/loadFontAsync.android.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { NativeModules } from 'react-native';
+let { ExponentFontLoader } = NativeModules;
+let _FONTS$ = {};
+
+/**
+ * Load a font asynchronously from a url, with a given `fontFamily`
+ * and `fontStyle`.
+ *
+ * NOTE: Unlike On iOS, on Android, we can't infer fontFamily and
```
Comment (brentvatne) [inline utils/loadFontAsync.android.js]: stupid small nit: "Unlike On" => "Unlike on"

File: utils/loadFontAsync.ios.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+let _FONTS$ = {};
+
+/**
+ * Load a font asynchronously from a url.
+ *
+ * NOTE: On iOS, we can infer the font family name and style
+ * from the downloaded font, so a URL is all we need to
+ * do the load.
+ */
+export default function loadFontAsync({ uri }) {
```
Comment (brentvatne) [inline utils/loadFontAsync.ios.js:12]: let's make sure we post a warning here if we are missing props that are required for it to work on Android @skevy (as per Slack discussion)

Comment (skevy) [inline utils/loadFontAsync.ios.js:12]: @brentvatne updated. see: https://github.com/exponentjs/exponent-with-custom-font/pull/1/files#diff-70455ac9c6ca2fdfb283f3d67eeb1332R66

**Pull request merged**